cli/cli/command_framework/highlevel/enclave_id_arg/enclave_id_arg.go

```diff
@@ -104,41 +105,45 @@ func getValidationFunc(argKey string, _ string, isGreedy bool) func(context.Cont
 			}
 			enclaveIdentifiersToValidate = enclaveIds
-		} else {
-			enclaveIdentifier, err := args.GetNonGreedyArg(argKey)
-			if err != nil {
-				return stacktrace.Propagate(err, "Expected a value for non-greedy arg '%v' but didn't find one", argKey)
-			}
-			enclaveIdentifiersToValidate = []string{enclaveIdentifier}
 		}
 
-		enclaves, err := kurtosisCtx.GetEnclaves(ctx)
-		if err != nil {
-			return stacktrace.Propagate(err, "An error occurred getting enclaves, which is necessary to check if the enclaves exist")
-		}
+		// only validate enclave identifer if this is a greedy enclave identifier arg
+		if len(enclaveIdentifiersToValidate) > 1 {
+			enclaves, err := kurtosisCtx.GetEnclaves(ctx)
+			if err != nil {
+				return stacktrace.Propagate(err, "An error occurred getting enclaves, which is necessary to check if the enclaves exist")
+			}
 
-		enclavesByShortenedUuid := enclaves.GetEnclavesByShortenedUuid()
-		enclavesByUuid := enclaves.GetEnclavesByUuid()
-		enclavesByName := enclaves.GetEnclavesByName()
+			enclavesByShortenedUuid := enclaves.GetEnclavesByShortenedUuid()
+			enclavesByUuid := enclaves.GetEnclavesByUuid()
+			enclavesByName := enclaves.GetEnclavesByName()
 
-		for _, enclaveIdentifier := range enclaveIdentifiersToValidate {
-			if _, found := enclavesByUuid[enclaveIdentifier]; found {
-				continue
-			}
-			if matches, found := enclavesByShortenedUuid[enclaveIdentifier]; found {
-				if len(matches) > validShortenedUuidOrNameMatches {
-					return stacktrace.NewError("Found multiple matching uuids '%v' for shortened uuid '%v' which is ambiguous", enclaveInfosToUuidsStr(matches), enclaveIdentifier)
+			for _, enclaveIdentifier := range enclaveIdentifiersToValidate {
+				if _, found := enclavesByUuid[enclaveIdentifier]; found {
+					continue
 				}
-				continue
-			}
-			if matches, found := enclavesByName[enclaveIdentifier]; found {
-				if len(matches) > validShortenedUuidOrNameMatches {
-					return stacktrace.NewError("Found multiple matching uuids '%v' for name '%v' which is ambiguous", enclaveInfosToUuidsStr(matches), enclaveIdentifier)
+				if matches, found := enclavesByShortenedUuid[enclaveIdentifier]; found {
+					if len(matches) > validShortenedUuidOrNameMatches {
+						return stacktrace.NewError("Found multiple matching uuids '%v' for shortened uuid '%v' which is ambiguous", enclaveInfosToUuidsStr(matches), enclaveIdentifier)
+					}
+					continue
 				}
-				continue
+				if matches, found := enclavesByName[enclaveIdentifier]; found {
+					if len(matches) > validShortenedUuidOrNameMatches {
+						return stacktrace.NewError("Found multiple matching uuids '%v' for name '%v' which is ambiguous", enclaveInfosToUuidsStr(matches), enclaveIdentifier)
+					}
+					continue
+				}
+				return stacktrace.NewError("No enclave found for identifier '%v'", enclaveIdentifier)
+			}
+		} else if len(enclaveIdentifiersToValidate) == 1 {
```

Comment:
ah, you answer this here. I'd recommend putting this case first with a comment, so it's clearer what's going on